explore/src/main/scala/explore/config/ITCRequests.scala

```diff
@@ -69,64 +79,68 @@ object ITCRequests {
         InstrumentModes(m.toGmosNITCInput)
       }
       // Discard values in the cache
-      .filterNot { case m =>
-        cache.get.cache.contains((wavelength, signalToNoise, constraints, m))
+      .filterNot { case mode =>
+        val params = ITCRequestParams(wavelength, signalToNoise, constraints, targets, mode)
```

Comment:
We may want to map into `ITCRequestParams` here and make `itcRows` a `List[ITCRequestParams]`. Otherwise we are creating it twice (once here and once in line 92).